### PR TITLE
Clean up hidden type registration

### PR DESCRIPTION
**Pull request by oli-obk**
work on https://github.com/rust-lang/rust/issues/101186

Actually passing down the relation and using it instead of `eq` for the hidden type comparison has *no* effect whatsoever and allows for no further improvements at the call sites. I decided the increased complexity was not worth it and thus did not include that change in this PR.

r? @compiler-errors

### CODE REVIEW

File: compiler/rustc_infer/src/infer/nll_relate/mod.rs [hunk 1]
```diff
@@ -395,12 +394,10 @@ where
 
     fn relate_opaques(&mut self, a: Ty<'tcx>, b: Ty<'tcx>) -> RelateResult<'tcx, Ty<'tcx>> {
         let (a, b) = if self.a_is_expected() { (a, b) } else { (b, a) };
+        let cause = self.delegate.cause().clone();
```
Comment (compiler-errors) [inline compiler/rustc_infer/src/infer/nll_relate/mod.rs:397]: why is this clone needed?

Comment (oli-obk) [reply compiler/rustc_infer/src/infer/nll_relate/mod.rs:397]: Ah that's just noise fromthe attempt to pass down the relation itself.

**Pull request merged**